### PR TITLE
Track recipe usage by token in the recipe  manager

### PR DESCRIPTION
**Pull request by Mekle001**
# Changes proposed in this pull request:
This enables tracking of all recipes used in the world, and filters the global recipe store by what is used during save. 

This will help keep colony.dat storage down to just what is necessary. 

This also clears the recipe list on world unload, so that the recipe list doesn't bleed over between worlds when switching between single player worlds. 

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/requestsystem/management/manager/StandardRecipeManager.java [hunk 1]
```diff
@@ -96,4 +111,20 @@ public void read(@NotNull final CompoundNBT compound)
...
+        recipes.clear();
+        usedRecipes.clear();
+    }
+
+    @Override
+    public void registerUse(final IToken<?> token)
+    {
+        if(!usedRecipes.contains(token))
+        {
+            usedRecipes.add(token);
```
Comment (MatthiasMann) [inline src/main/java/com/minecolonies/coremod/colony/requestsystem/management/manager/StandardRecipeManager.java]: No need to check contains - add already does that.

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/requestsystem/management/manager/StandardRecipeManager.java]: ye it's a set, it shouldn't need contains

Comment (Mekle001) [reply src/main/java/com/minecolonies/coremod/colony/requestsystem/management/manager/StandardRecipeManager.java]: Fair enough. Should be fixed.

**Review by someaddons — APPROVED** [review]

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/ldtteam/minecolonies/pull-request/5941/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ldtteam/minecolonies)

**Review by marchermans — APPROVED** [review]

**Pull request merged**